### PR TITLE
Support custom base64 encoding in golang helper methods

### LINKED ISSUE #6: golang UnmarshalBase64() is not compatible with python marshalled file
golang UnmarshalBase64() would throw an exception while decoding a file marshaled using python.

You can reproduce the issue with the following line:
bloom, err := inbloom.UnmarshalBase64("j+gAZAAAAGQAAAAAAAAAAAAAAABAAAAAAAAAAAAAAAAAAAAIAAAAAEAAAAAAAAAAAAABAAAAAAAAAYAAAAAAACAAAAAAAAAAAAAAAAAAAAQAAAAAAAAAAAAAAAAAgAAAAAAAAAAAAAAAAAAAQAAgAAAAAAAAAAAAAAAAAACBgAA=")

I've experimented changing bloom.go:174 to:
	if b, err := base64.StdEncoding.DecodeString(b64); err != nil {

Using base64.StdEncoding seems to make it work.

### PR DESCRIPTION
**Pull request by bergundy**
Closes #6

### CODE REVIEW

**Comment by bergundy** [comment]
@dvirsky apparently I don't have write access to EverythingMe/inbloom.
PTAL

File: go/inbloom/bloom.go [hunk 1]
```diff
@@ -161,17 +161,29 @@ func (f *BloomFilter) Marshal() []byte {
...
 }
 
-// MarshalBase64 is a convenience method that dumps the filter's data to a base64 encoded string,
-// ready to be passed as an GET/POST parameter
-func (f *BloomFilter) MarshalBase64() string {
-	return base64.URLEncoding.EncodeToString(f.Marshal())
+// MarshalBase64 is a convenience method that dumps the filter's data to a base64 encoded string.
+// By default uses URLEncoding which ready to be passed as a GET/POST parameter.
+// Pass an encoding param to use different encoding.
+func (f *BloomFilter) MarshalBase64(encoding ...*base64.Encoding) string {
```
Comment (dvirsky) [inline go/inbloom/bloom.go:167]: I don't like this pattern so much. 
How about adding func (f *BloomFilter) MarshalBase64Encoding(encoding *base64.Encoding) and have the old function just call this one with nil?

File: go/inbloom/bloom.go [hunk 2]
```diff
@@ -161,17 +161,29 @@ func (f *BloomFilter) Marshal() []byte {
...
 }
 
-// UnmarshalBase64 is a convenience function that unmarshals a filter that has been encoded into
-// a url parameter
-func UnmarshalBase64(b64 string) (*BloomFilter, error) {
-
-	if b, err := base64.URLEncoding.DecodeString(b64); err != nil {
+// UnmarshalBase64 is a convenience function that unmarshals a filter that has been encoded into base64.
+// Uses URLEncoding by default, pass an encoding param to use different encoding.
+func UnmarshalBase64(b64 string, encoding ...*base64.Encoding) (*BloomFilter, error) {
```
Comment (dvirsky) [inline go/inbloom/bloom.go:179]: same here.

**Review by dvirsky — CHANGES_REQUESTED** [review]
If you feel strongly about keeping it that way just tell me and I'll approve.

**Comment by bergundy** [comment]
@dvirsky I like it better the way I did it. I'm merging so we can move forward.

**Pull request merged**